conformance/operator_test.go

```diff
@@ -265,6 +267,114 @@ var _ = Describe("operator", func() {
 				Expect(err).To(HaveOccurred())
 			})
 		})
+		Context("Resource Injector", func() {
+			// 25815
+			It("Should inject downward api volume", func() {
+				node := sriovInfos.Nodes[0]
+				intf, err := sriovInfos.FindOneSriovDevice(node)
+				Expect(err).ToNot(HaveOccurred())
+
+				nodePolicy := &sriovv1.SriovNetworkNodePolicy{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "apivolumepolicy",
+						Namespace:    operatorNamespace,
+					},
+
+					Spec: sriovv1.SriovNetworkNodePolicySpec{
+						NodeSelector: map[string]string{
+							"kubernetes.io/hostname": node,
+						},
+						NumVfs:       5,
+						ResourceName: "apivolresource",
+						Priority:     99,
+						NicSelector: sriovv1.SriovNetworkNicSelector{
+							PfNames: []string{intf.Name},
+						},
+						DeviceType: "netdevice",
+					},
+				}
+
+				err = clients.Create(context.Background(), nodePolicy)
+				Expect(err).ToNot(HaveOccurred())
+
+				Eventually(func() bool {
+					stable, err := cluster.SriovStable(operatorNamespace, clients)
+					Expect(err).ToNot(HaveOccurred())
+					return stable
+				}, 5*time.Minute, 1*time.Second).Should(Equal(true))
+
+				Eventually(func() int64 {
+					testedNode, err := clients.Nodes().Get(node, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					resNum, _ := testedNode.Status.Capacity["openshift.io/apivolresource"]
+					capacity, _ := resNum.AsInt64()
+					return capacity
+				}, 3*time.Minute, time.Second).Should(Equal(int64(5)))
+
+				sriovNetwork := &sriovv1.SriovNetwork{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "apivolnetwork",
+						Namespace: operatorNamespace,
+					},
+					Spec: sriovv1.SriovNetworkSpec{
+						ResourceName:     "apivolresource",
+						IPAM:             `{"type":"host-local","subnet":"10.10.10.0/24","rangeStart":"10.10.10.171","rangeEnd":"10.10.10.181","routes":[{"dst":"0.0.0.0/0"}],"gateway":"10.10.10.1"}`,
+						NetworkNamespace: namespaces.Test,
+					}}
+				err = clients.Create(context.Background(), sriovNetwork)
+				Expect(err).ToNot(HaveOccurred())
+
+				pod := &corev1.Pod{
```

Comment:
I think later you can use the pod package from this PR https://github.com/openshift/sriov-tests/pull/18

Reply:
Yes, after everything is settled we can do a round of refactoring